bit operations on constants

The following program `sample.go` triggers a panic:

```go
package main

const (
    numDec uint8 = (1 << iota) / 2
    numHex 
    numOct
    numFloat
)   

func main() {
    kind := numDec
    println(kind & (numHex | numOct))
}   
```

Expected result:
```console
$ go run ./sample.go
0
```

Got:
```console
$ yaegi ./sample.go
run: ./sample.go:12:18: CFG post-order panic: invalid binary operation 1 | 2
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0000f2a00, 0x40)
        /usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0001cbc70)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:480 +0xc5
panic(0xeafee0, 0xc0004f80a0)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2.1(0xc0004f3b00)
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:457 +0xb3
panic(0xdd1ae0, 0xc0001aadf0)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
go/constant.BinaryOp(0x1098aa0, 0xc000360900, 0x12, 0x1098aa0, 0xc000360940, 0x194, 0x0)
        /usr/lib/go/src/go/constant/value.go:1242 +0x15b7
github.com/traefik/yaegi/interp.orConst(0xc0004f3b00)
        /home/marc/go/src/github.com/traefik/yaegi/interp/op.go:567 +0x3e9
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2(0xc0004f3b00)
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:709 +0x9c5e
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3b00, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:220 +0xa5
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3a00, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3800, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3600, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3500, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f3100, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0004f2c00, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0000bc500, 0xc0001cbaf8, 0xc0001cbac8)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:217 +0x66
github.com/traefik/yaegi/interp.(*Interpreter).cfg(0xc0003ee000, 0xc0000bc500, 0xc0003ea860, 0x4, 0xc0003ea860, 0x4, 0x0, 0x0, 0xc0000bc500)
        /home/marc/go/src/github.com/traefik/yaegi/interp/cfg.go:63 +0x20f
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0003ee000, 0xc000358f00, 0x96, 0x7ffc93da49bc, 0x14, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:507 +0x2f3
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0003ee000, 0x7ffc93da49bc, 0x14, 0xc000358e00, 0x96, 0x0, 0x0, 0xc0003f04e0)
        /home/marc/go/src/github.com/traefik/yaegi/interp/interp.go:394 +0xfa
main.runFile(0xc0003ee000, 0x7ffc93da49bc, 0x14, 0xc00007ed80, 0x1)
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:123 +0xae
main.run(0xc00000e090, 0x1, 0x1, 0x13, 0x13)
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:89 +0x7e5
main.main()
        /home/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x415
```

--- a/_test/const22.go
+++ b/_test/const22.go
@@ -1,0 +1,15 @@
+package main
+
+const (
+	numDec uint8 = (1 << iota) / 2
+	numHex
+	numOct
+	numFloat
+)
+
+func main() {
+	println(13 & (numHex | numOct))
+}
+
+// Output:
+// 1

--- a/internal/cmd/genop/genop.go
+++ b/internal/cmd/genop/genop.go
@@ -209,7 +209,11 @@ func {{$name}}Const(n *node) {
 		v := constant.BinaryOp(vConstantValue(v0), operator, vConstantValue(v1))
 		n.rval.Set(reflect.ValueOf(v))
 		{{- else}}
+		{{- if $op.Int}}
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.{{tokenFromName $name}}, constant.ToInt(vConstantValue(v1)))
+		{{- else}}
 		v := constant.BinaryOp(vConstantValue(v0), token.{{tokenFromName $name}}, vConstantValue(v1))
+		{{- end}}
 		n.rval.Set(reflect.ValueOf(v))
 		{{- end}}
 	{{- if $op.Str}}
@@ -940,6 +944,7 @@ type Op struct {
 	Complex bool   // true if operator applies to complex
 	Shift   bool   // true if operator is a shift operation
 	Bool    bool   // true if operator applies to bool
+	Int     bool   // true if operator applies to int only
 }
 
 func main() {
@@ -968,17 +973,17 @@ func main() {
 	b := &bytes.Buffer{}
 	data := map[string]interface{}{
 		"Arithmetic": map[string]Op{
-			"add":    {"+", true, true, true, false, false},
-			"sub":    {"-", false, true, true, false, false},
-			"mul":    {"*", false, true, true, false, false},
-			"quo":    {"/", false, true, true, false, false},
-			"rem":    {"%", false, false, false, false, false},
-			"shl":    {"<<", false, false, false, true, false},
-			"shr":    {">>", false, false, false, true, false},
-			"and":    {"&", false, false, false, false, false},
-			"or":     {"|", false, false, false, false, false},
-			"xor":    {"^", false, false, false, false, false},
-			"andNot": {"&^", false, false, false, false, false},
+			"add":    {"+", true, true, true, false, false, false},
+			"sub":    {"-", false, true, true, false, false, false},
+			"mul":    {"*", false, true, true, false, false, false},
+			"quo":    {"/", false, true, true, false, false, false},
+			"rem":    {"%", false, false, false, false, false, true},
+			"shl":    {"<<", false, false, false, true, false, true},
+			"shr":    {">>", false, false, false, true, false, true},
+			"and":    {"&", false, false, false, false, false, true},
+			"or":     {"|", false, false, false, false, false, true},
+			"xor":    {"^", false, false, false, false, false, true},
+			"andNot": {"&^", false, false, false, false, false, true},
 		},
 		"IncDec": map[string]Op{
 			"inc": {Name: "+"},
@@ -996,7 +1001,7 @@ func main() {
 			"not":    {Name: "!", Float: false, Bool: true},
 			"neg":    {Name: "-", Float: true, Bool: false},
 			"pos":    {Name: "+", Float: true, Bool: false},
-			"bitNot": {Name: "^", Float: false, Bool: false},
+			"bitNot": {Name: "^", Float: false, Bool: false, Int: true},
 		},
 	}
 	if err = parse.Execute(b, data); err != nil {

--- a/interp/op.go
+++ b/interp/op.go
@@ -253,7 +253,7 @@ func andConst(n *node) {
 	n.rval = reflect.New(t).Elem()
 	switch {
 	case isConst:
-		v := constant.BinaryOp(vConstantValue(v0), token.AND, vConstantValue(v1))
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.AND, constant.ToInt(vConstantValue(v1)))
 		n.rval.Set(reflect.ValueOf(v))
 	case isUint(t):
 		n.rval.SetUint(vUint(v0) & vUint(v1))
@@ -338,7 +338,7 @@ func andNotConst(n *node) {
 	n.rval = reflect.New(t).Elem()
 	switch {
 	case isConst:
-		v := constant.BinaryOp(vConstantValue(v0), token.AND_NOT, vConstantValue(v1))
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.AND_NOT, constant.ToInt(vConstantValue(v1)))
 		n.rval.Set(reflect.ValueOf(v))
 	case isUint(t):
 		n.rval.SetUint(vUint(v0) &^ vUint(v1))
@@ -564,7 +564,7 @@ func orConst(n *node) {
 	n.rval = reflect.New(t).Elem()
 	switch {
 	case isConst:
-		v := constant.BinaryOp(vConstantValue(v0), token.OR, vConstantValue(v1))
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.OR, constant.ToInt(vConstantValue(v1)))
 		n.rval.Set(reflect.ValueOf(v))
 	case isUint(t):
 		n.rval.SetUint(vUint(v0) | vUint(v1))
@@ -799,7 +799,7 @@ func remConst(n *node) {
 	n.rval = reflect.New(t).Elem()
 	switch {
 	case isConst:
-		v := constant.BinaryOp(vConstantValue(v0), token.REM, vConstantValue(v1))
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.REM, constant.ToInt(vConstantValue(v1)))
 		n.rval.Set(reflect.ValueOf(v))
 	case isUint(t):
 		n.rval.SetUint(vUint(v0) % vUint(v1))
@@ -1195,7 +1195,7 @@ func xorConst(n *node) {
 	n.rval = reflect.New(t).Elem()
 	switch {
 	case isConst:
-		v := constant.BinaryOp(vConstantValue(v0), token.XOR, vConstantValue(v1))
+		v := constant.BinaryOp(constant.ToInt(vConstantValue(v0)), token.XOR, constant.ToInt(vConstantValue(v1)))
 		n.rval.Set(reflect.ValueOf(v))
 	case isUint(t):
 		n.rval.SetUint(vUint(v0) ^ vUint(v1))

